ui/app/src/components/PreviewBrowseComponentsPanel/PreviewBrowseComponentsPanel.tsx

```diff
@@ -179,7 +87,10 @@ export default function PreviewBrowseComponentsPanel() {
         const items =
           source.page[source.pageNumber]
             ?.map((id: string) => source.byId[id])
-            .filter((item: ContentInstance) => item.craftercms.contentTypeId === source.contentTypeFilter) || [];
+            .filter(
+              (item: ContentInstance) =>
+                source.contentTypeFilter === 'all' || item.craftercms.contentTypeId === source.contentTypeFilter
```

Comment:
- too inefficient to iterate through the whole array when you know in advanced that you needn't apply the filter at all
- also, the `||` is misplaced; it doesn't avoid an exception if the .map returns undefined
```js
let items = source.page[source.pageNumber]?.map(...) ?? []
if (source.contenTypeFilter !== 'all') 
  items = items.filter(...)